cvr/src/main/java/org/cbioportal/cmo/pipelines/cvr/seg/CVRSegFieldSetMapper.java

```diff
@@ -45,22 +45,44 @@
 import org.springframework.validation.BindException;
 
 public class CVRSegFieldSetMapper implements FieldSetMapper<CVRSegRecord> {
-    Logger log = Logger.getLogger(CVRSegFieldSetMapper.class);
+    static Logger log = Logger.getLogger(CVRSegFieldSetMapper.class);
 
     @Override
     public CVRSegRecord mapFieldSet(FieldSet fs) throws BindException {
         CVRSegRecord record = new CVRSegRecord();
         List<String> fields = CVRSegRecord.getFieldNames();
         for (int i = 0; i < fields.size(); i++) {
             String field = fields.get(i);
-            try {
-                record.getClass().getMethod("set" + field, String.class).invoke(record, fs.readString(i));
-            } catch (Exception e) {
-                if (e.getClass().equals(NoSuchMethodException.class)) {
-                    log.info("No set method exists for " + field);
-                }
-            }
+            String value = (i < fs.getFieldCount()) ? fs.readString(i) : "";  // default to empty string if out of bounds
+            setFieldValue(record, field, value);
```

Comment:
I'm losing count of how many times getClass().getMethod() had been called...and now dropped!